src/rail.cpp

```diff
@@ -19,6 +19,18 @@
 
 #include "safeguards.h"
 
+/**
+ * Get the RailType for this RailTypeInfo.
+ * @return RailType in static RailTypeInfo definitions.
+ */
+RailType RailTypeInfo::Index() const
+{
+	extern RailTypeInfo _railtypes[RAILTYPE_END];
+	size_t index = this - _railtypes;
+	assert(index < RAILTYPE_END && this == _railtypes + index);
```

Comment:
I know it was there already, but the second part of this assert seems... unnecessary

Also for the other function